Dockerfile

```diff
@@ -1,3 +1,74 @@
+##
+# Build Stage
+#
+FROM ubuntu:focal as build
+
+ENV KEYSTORE_PW="pass:kspass"
+ENV TRUSTSTORE_PW="tspass"
+ENV DEFAULT_PW="oepass"
+ENV INSTALLER_CREATION_DIR="OEInstaller"
+ENV STAGING_DIR="OEInstaller_stagingDir"
+ENV OE_BRANCH="master"
+
+##
+# Prerequesites
+#
+RUN apt-get update && apt-get upgrade && \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y \
+      openssl net-tools python default-jdk maven \ 
+      apache2-utils git apt-transport-https \
+      ca-certificates curl gnupg lsb-release software-properties-common\
+    && apt-get clean
+
+##
+# Certificates
+#
+
+# Self-signed Certs
+RUN openssl req -x509 -nodes -days 365 -newkey rsa:2048 -keyout /etc/ssl/private/apache-selfsigned.key -out /etc/ssl/certs/apache-selfsigned.crt \ 
+    -subj "/C=US/ST=WA/L=Seattle/O=I-TECH-UW/OU=DIGI/CN=localhost"
+
+RUN mkdir /etc/openelis-global/
+# Keystore
+RUN openssl pkcs12 -inkey /etc/ssl/private/apache-selfsigned.key -in /etc/ssl/certs/apache-selfsigned.crt -export -out /etc/openelis-global/keystore --passin ${KEYSTORE_PW} --passout ${KEYSTORE_PW}
+# # Client-facing Keystore
+RUN cp /etc/openelis-global/keystore /etc/openelis-global/client_facing_keystore
+# # Truststore
+RUN keytool -import -alias oeCert -file /etc/ssl/certs/apache-selfsigned.crt -storetype pkcs12 -keystore /etc/openelis-global/truststore -storepass ${TRUSTSTORE_PW} -noprompt
+
+##
+# Copy Source Code
+#
+ADD ./pom.xml /build/pom.xml
+ADD ./tools /build/tools
+ADD ./src /build/src
+ADD ./install /build/install
+ADD ./dev /build/dev
+
+WORKDIR /build
+
+##
+# Checkout Dependencies
+#
+RUN git clone https://github.com/I-TECH-UW/dataexport.git --branch develop
```

Comment:
We should probably have an env var for which branch / tag to use for each of these dependencies

Reply:
Could this be read from the .gitmodules file? or can we just copy the files in from the submodule and ensure that the project is cloned with the --recurse-submodules option? Trying to cut down on having 2 locations that could point to different branches/tag